Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -405,20 +576,9 @@ namespace AZ::DocumentPropertyEditor
             if (parentContainerAttribute && !parentContainerAttribute->IsNull() &&
                 parentContainerInstanceAttribute && !parentContainerInstanceAttribute->IsNull())
             {
-                auto parentContainer = AZ::Dom::Utils::ValueToTypeUnsafe<AZ::SerializeContext::IDataContainer*>(*parentContainerAttribute);
-                auto parentContainerInstance = AZ::Dom::Utils::ValueToTypeUnsafe<void*>(*parentContainerInstanceAttribute);
-
-                // check if this element is actually standing in for a direct child of a container. This is used in scenarios like
-                // maps, where the direct children are actually pairs of key/value, but we need to only show the value as an editable item
-                // who pretends that they can be removed directly from the container
-                auto containerElementOverrideAttribute = attributes.Find(AZ::Reflection::DescriptorAttributes::ContainerElementOverride);
-                if (containerElementOverrideAttribute && !containerElementOverrideAttribute->IsNull())
-                {
-                    instance = AZ::Dom::Utils::ValueToTypeUnsafe<void*>(*containerElementOverrideAttribute);
-                }
-
-                m_containers.SetValue(m_builder.GetCurrentPath(), BoundContainer{ parentContainer, parentContainerInstance, instance });
+                m_containers.SetValue(m_builder.GetCurrentPath(), AZStd::move(BoundContainer::CreateBoundContainer(instance, attributes)));
```

Comment:
The AZStd::move is unnecessary as CreateBoundContainer returns temporary value, so it would bind to an rvalue reference regardless.
```suggestion
                m_containers.SetValue(m_builder.GetCurrentPath(), BoundContainer::CreateBoundContainer(instance, attributes));
```

Reply:
Removed the AZStd::move.